types/ember-data/test/transform.ts

```diff
@@ -15,8 +16,41 @@ class PointTransform extends DS.Transform {
     }
 }
 
+class EnumTransform extends DS.Transform {
+    deserialize<AllowedValue>(
+        serialized: any,
+        options: { allowedValues: Record<string, AllowedValue> },
+    ): AllowedValue {
```

Comment:
Conventionally, the type parameter here should just be `T`; while using a longer name can sometimes be informative it had me second-guessing whether I was misreading the type here.

```suggestion
    deserialize<T>(
        serialized: any,
        options: { allowedValues: Record<string, T> },
    ): T {
```